riscv-asm.md

```diff
@@ -19,7 +19,85 @@ duplicating it here.
 
 # Registers
 
-ISA and ABI register names for X, F, and CSRs.
+Registers are the most important part of any processor. RISC-V defines various
+types, depending on which extensions are included: The general registers (with
+the program counter), control registers, floating point registers (F extension),
+and vector registers (V extension).
+
+## General registers 
+
+The RV32I base integer ISA includes 32 registers, named `x0` to `x31`. The
+program counter `PC` is separate from these registers, in contrast to other
+processors such as the ARM-32. The first register, `x0`, has a special function:
+Reading it always returns 0 and writes to it are ignored. As we will see later,
+this allows various tricks and simplifications.
+
+In practice, the programmer doesn't use this notation for the registers. Though
+`x1` to `x31` are all equally general-use registers as far as the processor is
+concerned, by convention certain registers are used for special tasks. The
+RISC-V **application binary interface** (ABI) standardizes this by giving the
+registers other "names" or "calling conventions". These are recognized by the
```

Comment:
It's not really accurate to say the registers have calling conventions, it's the language that has a calling convention.